### PR TITLE
Feat: Delete resource by labels

### PR DESCRIPTION
**Pull request by wangyikewxgm**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/providers/kube/handle.go [hunk 1]
```diff
@@ -263,6 +265,7 @@ func (h *provider) List(ctx monitorContext.Context, wfCtx wfContext.Context, v *
 
 // Delete deletes CR from cluster.
 func (h *provider) Delete(ctx monitorContext.Context, wfCtx wfContext.Context, v *value.Value, act types.Action) error {
+	fmt.Println(v.CueValue())
```
Comment (FogDong) [inline pkg/providers/kube/handle.go:268]: debug log?

File: pkg/providers/kube/handle.go [hunk 2]
```diff
@@ -263,6 +265,7 @@ func (h *provider) List(ctx monitorContext.Context, wfCtx wfContext.Context, v *
 
 // Delete deletes CR from cluster.
 func (h *provider) Delete(ctx monitorContext.Context, wfCtx wfContext.Context, v *value.Value, act types.Action) error {
```
Comment (FogDong) [inline pkg/providers/kube/handle.go:267]: Please add some tests.

File: pkg/providers/kube/handle.go [hunk 3]
```diff
@@ -276,9 +278,31 @@ func (h *provider) Delete(ctx monitorContext.Context, wfCtx wfContext.Context, v
 		return err
 	}
 	deleteCtx := handleContext(ctx, cluster)
+
+	filterValue, err := v.LookupValue("filter")
+	if err != nil && !strings.Contains(err.Error(), "failed to lookup value: var(path=filter) not exist"){
```
Comment (FogDong) [inline pkg/providers/kube/handle.go:283]: You can ignore the err handle here like:

```
if filterValue, err := v.LookupValue("filter"); err == nil {
filter := &filters{}
if err := filterValue.UnmarshalTo(filter); err != nil {
			return err
		}
		labelSelector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: filter.MatchingLabels})
		if err != nil {
			return err
		}
		if err := h.cli.DeleteAllOf(deleteCtx , obj, &client.DeleteAllOfOptions{ListOptions: client.ListOptions{Namespace: filter.Namespace, LabelSelector:  labelSelector}}); err != nil {
			return v.FillObject(err.Error(), "err")
		}
		return nil
}
```

Comment (wangyikewxgm) [reply pkg/providers/kube/handle.go]: fixed

**Review by FogDong — APPROVED** [review]

**Pull request merged**